### PR TITLE
HS-202: Notification service with skaffold

### PR DESCRIPTION
**Pull request by james8hut**
## Description
New Notification service running in skaffold

## Jira link(s)
- https://issues.opennms.org/browse/HS-202

## Flagged for review
Spring boot application with a single hard coded REST method
To test run:
curl -i http://localhost:38080/notifications/pagerDutyKey

## Checklist
* [ ] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: notifications/src/test/java/org/opennms/horizon/notifications/NotificationsServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.notifications;
```
Comment (yangli2014) [inline notifications/src/test/java/org/opennms/horizon/notifications/NotificationsServiceTest.java:29]: Usually, it is better to keep the test class in the same java package as the tested class.

**Review by yangli2014 — COMMENTED** [review]
Nice job. I think the notification Spring Boot application hasn't been finished yet. The scaffold change is working great.

File: notifications/src/test/java/org/opennms/horizon/notifications/NotificationsServiceTest.java (hunk 1 above)
Comment (james8hut) [reply notifications/src/test/java/org/opennms/horizon/notifications/NotificationsServiceTest.java]: Fixed!

File: notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationsService.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.notifications.service;
+
+import org.springframework.stereotype.Service;
+
+@Service
+public class NotificationsService implements INotificationsService {
```
[the remark this answers is not in the document]
Comment (james8hut) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/NotificationsService.java]: Fixed!

File: notifications/src/main/java/org/opennms/horizon/notifications/service/INotificationsService.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.horizon.notifications.service;
+
+public interface INotificationsService {
```
Comment (yangli2014) [inline notifications/src/main/java/org/opennms/horizon/notifications/service/INotificationsService.java:31]: I suggest changing the name to NotificationService to keep consistency.

Comment (james8hut) [reply notifications/src/main/java/org/opennms/horizon/notifications/service/INotificationsService.java]: Done

**Review by yangli2014 — APPROVED** [review]

**Review by geraldhumphries — APPROVED** [review]

**Pull request merged**